meson.build

```diff
@@ -7,7 +7,7 @@ project(
   default_options: [
     'buildtype=debugoptimized',
     'c_std=c99',
-    'cpp_std=c++14',
+    'cpp_std=c++11',
```

Comment:
O_o, some compiler not supporting C++14?

Reply:
This is quick temporary hack to make sure the manylinux wheel tests are passing with this change. My hunch is that there were some `--std=c++11` in `setup.py` that were not carried over to `meson.build` files.

Previously there was a missing symbol for some reason I don't fully understand:

```
============================= test session starts ==============================
platform linux -- Python 3.12.1, pytest-8.0.2, pluggy-1.4.0
rootdir: /tmp/tmp.vRc1i4Drvm/test_cwd
collected 4487 items / 177 errors

==================================== ERRORS ====================================
_________ ERROR collecting cluster/_hdbscan/tests/test_reachibility.py _________
ImportError while importing test module '/tmp/tmp.vRc1i4Drvm/venv/lib/python3.12/site-packages/sklearn/cluster/_hdbscan/tests/test_reachibility.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/python/cp312-cp312/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../venv/lib/python3.12/site-packages/sklearn/cluster/__init__.py:6: in <module>
    from ._affinity_propagation import AffinityPropagation, affinity_propagation
../venv/lib/python3.12/site-packages/sklearn/cluster/_affinity_propagation.py:16: in <module>
    from ..metrics import euclidean_distances, pairwise_distances_argmin
../venv/lib/python3.12/site-packages/sklearn/metrics/__init__.py:7: in <module>
    from . import cluster
../venv/lib/python3.12/site-packages/sklearn/metrics/cluster/__init__.py:25: in <module>
    from ._unsupervised import (
../venv/lib/python3.12/site-packages/sklearn/metrics/cluster/_unsupervised.py:23: in <module>
    from ..pairwise import _VALID_METRICS, pairwise_distances, pairwise_distances_chunked
../venv/lib/python3.12/site-packages/sklearn/metrics/pairwise.py:43: in <module>
    from ._pairwise_distances_reduction import ArgKmin
../venv/lib/python3.12/site-packages/sklearn/metrics/_pairwise_distances_reduction/__init__.py:94: in <module>
    from ._dispatcher import (
../venv/lib/python3.12/site-packages/sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py:17: in <module>
    from ._argkmin_classmode import (
E   ImportError: /tmp/tmp.vRc1i4Drvm/venv/lib/python3.12/site-packages/sklearn/metrics/_pairwise_distances_reduction/_argkmin_classmode.cpython-312-x86_64-linux-gnu.so: undefined symbol: _ZdlPv
```

Reply:
Have you tried to use `'cpp_std=c++17'`?

Reply:
Same issue with `cpp_std=c++17`